app/javascript/controllers/autonumeric_controller.js

```diff
@@ -4,6 +4,7 @@ import AutoNumeric from "autonumeric"
 // Connects to data-controller="autonumeric"
 export default class extends Controller {
   connect() {
-    this.anElement = new AutoNumeric(this.element)
+    const options = this.element.dataset.autonumericOptions ? JSON.parse(this.element.dataset.autonumericOptions) : {};
+    this.anElement = new AutoNumeric(this.element, options)
```

Comment:
It may be better to leverage Stimulus' values here. It'll handle the parsing and the defaults for us.

```javascript
static values = {
    options: Object
}

connect() {
  this.anElement = new AutoNumeric(this.element, this.optionsValue)
}
```

Then in the html it would be `data-autonumeric-options-value="{}"`